include/exiv2/config.h

```diff
@@ -93,13 +93,15 @@ typedef int pid_t;
 #endif
 //////////////////////////////////////
 
-// https://softwareengineering.stackexchange.com/questions/291141/how-to-handle-design-changes-for-auto-ptr-deprecation-in-c11
-#if __cplusplus >= 201103L
-  #include <memory>
-  #include <sys/types.h>
-  #include <unistd.h>
-  template <typename T>
-  using auto_ptr = std::unique_ptr<T>;
+#ifndef  _MSC_VER
```

Comment:
We do not want to use this trick anymore when using MSVC ?

Reply:
The compiler went crazy.  MSVC does has neither <memory> nor <unistd>.  So, I tried hiding this from MSVC and that worked!

Reply:
I've modified that code.  MSVC mustn't include <unistd.h>, however the `using magic` is working.